Fix: check for JsonWebTokenError

alexanderleegs opened the pull request:
## Problem

This PR addresses an issue with our identification of expired tokens - previously, we were expecting a `TokenExpiredError` for expired tokens, however, we were receiving an abnormal number of unhandled server errors. The reason for this is that we also set an expiry for the cookie the token is attached to - this means that after the token/cookie expires, the token we attempt to decrypt is empty, which results in a `JsonWebTokenError` instead. As users getting logged out due to expiring tokens is an expected error, this PR handles this error in the same way, in order to reduce the number of alerts we receive on our monitoring channels.

seaerchin commented:
just double checking - if people attempt to change cookie and login etc, will we miss that? main worry is that changing the error type to cover `JsonWebTokenError ` might lead to actual useful logs being omitted.

if you don't feel that this is a big issue, i'm ok with this as-is.

seaerchin reviewed (commented):
as above

alexanderleegs commented:
> JsonWebTokenError

That's a great point actually, thanks for catching! We should still monitor when people are attempting to change their cookie - I've changed this check to just look for an empty cookie early. I've also added additional details to the errors that trigger our alarms by logging `err.message`. https://github.com/isomerpages/isomercms-backend/pull/423/commits/707dc3d0c7b03df47ae981145e98a4553c08a9a4

seaerchin reviewed (approved):
the check for `!isomercms` is dangerous - if `isomercms` is an _empty_ object, this will return `false`. i'm not too sure of the type so i'm approving this **only if** it's not an empty object.

seaerchin commented on services/middlewareServices/AuthMiddlewareService.js:
```diff
@@ -77,7 +81,7 @@ class AuthMiddlewareService {
         throw new AuthError(`JWT token has expired`)
       } else {
-        logger.error(`Authentication error. Url: ${url}`)
+        logger.error(`Authentication error. Message:${err.message} Url: ${url}`)
```
also extremely minor nit: could we add a space like so `Message: ${err.message}`

alexanderleegs replied:
Fixed in https://github.com/isomerpages/isomercms-backend/pull/423/commits/68574b98a088437c7edca179a1930164b22a797c

alexanderleegs commented:
> the check for `!isomercms` is dangerous - if `isomercms` is an _empty_ object, this will return `false`. i'm not too sure of the type so i'm approving this **only if** it's not an empty object.

Cookies are in a key:value format, so if the cookie is missing, `isomercms` will be undefined!
